### ISSUE TITLE
[PR WELCOME] Set language level for library sources (es5, es6, es7, dom, webworker, ...)

### ISSUE THREAD

**Issue by JulienOuvrard** [dherges/ng-packagr#162]
## Type of Issue

- [ ] Bug Report
- [x] Feature Request

## Description

I have made an application using a custom component library that I have implemented, and now i wanted to use ng-packagr to externalize my custom component library.
The application was created by angular-cli, and i have followed the tutorial of Nikolas LeBlanc and the structure shown in the ng-packaged project.

For some reasons i have used in development es7 features like includes on arrays and padStart on strings. 
With the angular-cli builds including polyfills of core-js everything was working fine but the ng-packagr build gave me errors like this 
```
Property 'includes' does not exist on type 'any[]'.
```

I have tried to include them as externals and import them in the components that required the functions but still have theses errors.

So, how can i add polyfills functions on the build ?

#### Version Information

```
ng-packagr: v1.3.0
node: v7.10.0
@angular: v4.1.3
rxjs: v5.4.2
zone.js: v0.8.14
core-js: v2.4.1
```

**Comment by dherges** [comment]
Hi @JulienOuvrard,

it seems to be not possible at the moment as the [auto-configuration for the TypeScript compiler is `dom`, `es2015` API](https://github.com/dherges/ng-packagr/blob/master/src/lib/conf/tsconfig.ngc.json#L26-L29).

What is the current best practice / recommendation for polyfills? Is es7 (es2016) already released "to go at it"? I remember there was a discussion on the angular main repo saying "we don't want to use ES7 yet"

**Comment by JulienOuvrard** [comment]
Hi @dherges,

Thanks for the answer, i'll manage to avoid the usage of es7 features for the moment to get my library compiled. (I've already made changes to avoid includes and padStart)

I really don't know the state of that practice, i need to make further research on it.

Here's an hypothesis, can we include a tsconfig in custom libraries ? In that way, ng-packagr will use that configuration instead of the auto-configuration.

**Comment by dherges** [comment]
Quick change at angular: they still use [lib: ["es2015", "dom"]](https://github.com/angular/angular/blob/master/packages/tsconfig-build.json#L16)

**Comment by dherges** [comment]
If wanted, the setting should be included in `ng-package.json`, e.g.:

```json
{
  "lib": {
    "target": "es7" /** values: "es5", "es6", "es7" **/
    /** other possible names: "runtime", "environment", ... */
  }
}
```

And from there, we would write it to the generated tsconfig.json file and then (I think) the assumption about the target environment should be somewhere written in package.json (?)

**Comment by JulienOuvrard** [comment]
I think that would be nice to have this feature in future release of ng-packagr.
The idea of the assumption about the target environment written in package.json seems good to me considering that it describe the package.

**Comment by DavidParks8** [comment]
Why duplicate the lib config in `ng-package.json` and `tsconfig.json`? It should be possible to search for and read the developer's `tsconfig.json` dynamically to determine what libs should be included. This could be accompanied by a field to specify a custom tsconfig file such as:
```json
{
  "tsconfig": "./../tsconfig.app.json"
}
```
It would also be nice to dynamically search .angular-cli.json for the tsconfig.

**Comment by fr0** [comment]
What is the workaround for this? If there is no workaround, I can't use `ng-packagr` until it is implemented.

**Comment by dherges** [comment]
Closed by #486

**Comment by tommck** [comment]
So, was there a reason I missed for not just pointing this to a tsconfig.json file?
